src/main/java/io/xpring/xrpl/DefaultXpringClient.java

```diff
@@ -85,7 +87,14 @@ public BigInteger getBalance(final String xrplAccountAddress) throws XpringKitEx
      * @return The status of the given transaction.
      */
     public TransactionStatus getTransactionStatus(String transactionHash) throws XpringKitException {
-        throw XpringKitException.unimplemented;
+        RawTransactionStatus transactionStatus = getRawTransactionStatus(transactionHash);
+
+        // Return PENDING if the transaction is not validated.
+        if (!transactionStatus.getValidated()) {
+            return TransactionStatus.PENDING;
+        }
+
+        return transactionStatus.getTransactionStatusCode().startsWith("tes") ? TransactionStatus.SUCCEEDED : TransactionStatus.FAILED;
```

Comment:
So this is not implementing the `UNKNOWN` for transactions we can't handle like Escrow, correct?

Reply:
This is correct. 

This is tracked in a separate Asana task and we decided this was okay to launch as is (since you can only generate payments here). [Here's the tracking task](https://app.asana.com/0/1143722888069470/1159282373720075).

I plan to follow up on this in the next sprint.